releasenotes/notes/add-name-field-targetentry-1643769a9bf3554d.yaml

```
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    A new field `name` is added to :c:struct:`QkTargetEntry` to hold an optional name for the target entry.
+    This will allow us to pass a name to gates with fixed parameters.
```

A couple of suggestions here:
The category should not be under ``features``, ``but features_c``:

```suggestion
features_c:
  - |
    A new field ``name`` is added to :c:struct:`QkTargetEntry` to hold an optional name for the target entry.
    This will allow us to pass a name to gates with fixed parameters.
```

There should also be a section under the correct ``upgrade`` category explaining the functions were affected by this change. Since this breaks existing API it is necessary for us to correctly document this.

---

Thanks @raynelfss. I have updated the release notes with the said changes and an `upgrade` category.